src/Symfony/Component/VarDumper/VarDumper.php

```diff
@@ -76,11 +77,11 @@ private static function register(): void
             case 'server' === $format:
             case $format && 'tcp' === parse_url($format, \PHP_URL_SCHEME):
                 $host = 'server' === $format ? $_SERVER['VAR_DUMPER_SERVER'] ?? '127.0.0.1:9912' : $format;
-                $dumper = \in_array(\PHP_SAPI, ['cli', 'phpdbg', 'embed'], true) ? new CliDumper() : new HtmlDumper();
+                $dumper = self::guessMostSuitableDumper();
```

Comment:
let's make things simpler, same below:
```suggestion
                $dumper = str_contains($_SERVER['HTTP_ACCEPT'] ?? 'html', 'html')  ? new HtmlDumper() : new CliDumper();
```